apport/packaging_impl/apt_dpkg.py

```diff
@@ -1909,33 +1909,15 @@ def _build_apt_sandbox(
                     distro=distro_name,
                     ppa_name=urllib.parse.quote(ppa_name),
                 )
-                ppa_info = self.json_request(ppa_archive_url)
-                if not ppa_info:
-                    continue
-                try:
-                    signing_key_fingerprint = ppa_info["signing_key_fingerprint"]
-                except IndexError:
-                    apport.logging.warning(
-                        "Error: can't find signing_key_fingerprint at %s",
-                        ppa_archive_url,
-                    )
+                ppa_key = self.json_request(
+                    ppa_archive_url + "?ws.op=getSigningKeyData"
+                )
+                if not ppa_key:
                     continue
-                argv = [
-                    "apt-key",
-                    "--keyring",
-                    os.path.join(trusted_d, f"{origin}.gpg"),
-                    "adv",
-                    "--quiet",
-                    "--keyserver",
-                    "keyserver.ubuntu.com",
-                    "--recv-key",
-                    signing_key_fingerprint,
-                ]
-
-                if subprocess.call(argv) != 0:
-                    apport.logging.warning(
-                        "Unable to import key for %s", ppa_archive_url
-                    )
+                with open(
+                    os.path.join(trusted_d, f"{origin}.asc"), "w", encoding="utf-8"
+                ) as key_out:
+                    key_out.write(ppa_key)
```

Comment:
You could use pathlib to do the same with a little less code. Should look like this (more or less):
```suggestion
                key_file = pathlib.Path(trusted_d) / f"{origin}.asc"
                key_file.write_text(ppa_key, encoding="utf-8")
```